### ISSUE TITLE
DXVK support

### ISSUE THREAD

**Issue by Nemirtingas** [Nemirtingas/ingame_overlay#21]

**Comment by Sue1133** [comment]
Is this planned?

**Comment by Nemirtingas** [comment]
It is a desirable feature. But I encounter many issues hooking the Vulkan functions. The way to go is to insert a layer in the Vulkan chain, but I didn't set my mind on how I would do it. There is a branch for Vulkan support that kind of works, but its not ready for daily use.

**Comment by Sue1133** [comment]
Thanks for the explanation. The strange thing is that in dx12 games via vkd3d-proton overlay works fine.

**Comment by Nemirtingas** [comment]
Yes, because the DX12 hook works

**Comment by Sue1133** [comment]
Thank you very much for your work. Of course I can't check the work in DXVK now, but I hope I will be able to in the near future.

**Comment by YoteZip** [comment]
Isn't this the same overlay that was used in the older Goldberg Steam emulator? The older overlay version continues to work on nearly everything through DXVK/Proton (aside from causing crashes in most DX9 games). The newer version via the Detanup fork doesn't seem to be doing as well in comparison - is this possibly a regression?

**Comment by Nemirtingas** [comment]
I am not a developper of Goldberg Steam emulator. Please provide a reproductible issue.
Also, DXVK is not officially supported, so there's nothing I can do currently until I handle this feature.

**Comment by YoteZip** [comment]
Right, just saying that if this were attempted to be added/fixed, the version of this code that lives at [the old GSE repo](https://gitlab.com/Mr_Goldberg/goldberg_emulator/-/tree/master/overlay_experimental?ref_type=heads) does seem to be set up in a way that allows DXVK to work almost across the board - I'm not sure if this is something Mr. Goldberg did intentionally and didn't push upstream, or if it's coincidental.

**Comment by ndiruhniu** [comment]
Thanks, it works now

**Comment by Nemirtingas** [comment]
I never added DXVK support. The only thing that has been added was DXVK detection so DX9, DX10 and DX11 shows (DXVK) in their title.

**Comment by ndiruhniu** [comment]
Ok, but the overlay works now.

**Comment by Nemirtingas** [comment]
Latest version should work with DXVK 2.7.1. At least, when I tested, it did work. Didn't try on proton/wine, only on Windows.

**Comment by ndiruhniu** [comment]
Does not compile in gbe_fork(

**Comment by Nemirtingas** [comment]
I am not the developper of gbe_fork, create an issue there. It compiles just fine as you can see here: https://github.com/Nemirtingas/ingame_overlay/actions/runs/21886989982
